src/systemcmds/motor_ramp/motor_ramp.cpp

```diff
@@ -149,66 +152,111 @@ Note: this command currently only supports the `/dev/pwm_output0` output.
  */
 int motor_ramp_main(int argc, char *argv[])
 {
-	if (argc < 4) {
-		usage("missing parameters");
-		return 1;
-	}
+	int myoptind = 1;
+	int ch;
+	const char *myoptarg = nullptr;
+	bool error_flag = false;
+	bool set_pwm_min = false;
+	_max_pwm = 2000;
+	_ramp_time = 1.0f;
 
 	if (_thread_running) {
 		PX4_WARN("motor_ramp already running\n");
 		/* this is not an error */
 		return 0;
 	}
 
-	if (!strcmp(argv[1], "ramp")) {
-		_mode = RAMP;
+	if (argc < 4) {
+		usage("missing parameters");
+		return 1;
+	}
 
-	} else if (!strcmp(argv[1], "sine")) {
-		_mode = SINE;
+	while ((ch = px4_getopt(argc, argv, "d:a:b:r:", &myoptind, &myoptarg)) != EOF) {
+		switch (ch) {
 
-	} else if (!strcmp(argv[1], "square")) {
-		_mode = SQUARE;
+			case 'd':
+				if(!strcmp(myoptarg, "/dev/pwm_output0") || !strcmp(myoptarg, "/dev/pwm_output1")){
+					_pwm_output_dev = const_cast<char*>(myoptarg);
```

Comment:
The cast should not be required.
```suggestion
					_pwm_output_dev = myoptarg;
```